src/main/java/com/synopsys/integration/alert/web/config/ConfigController.java

```diff
@@ -27,52 +27,64 @@
 
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
 
 import com.synopsys.integration.alert.common.enumeration.ConfigContextEnum;
 import com.synopsys.integration.alert.web.controller.BaseController;
 import com.synopsys.integration.alert.web.model.FieldModel;
 
+@RestController
 @RequestMapping(ConfigController.CONFIGURATION_PATH)
 public class ConfigController extends BaseController {
     public static final String CONFIGURATION_PATH = BaseController.BASE_PATH + "/configuration";
 
     private final ConfigControllerHandler controllerHandler;
 
-    // TODO Change the endpoint to have configuration/context/type/descriptorName/id
-
     @Autowired
     public ConfigController(final ConfigControllerHandler controllerHandler) {
         this.controllerHandler = controllerHandler;
     }
 
-    public List<FieldModel> getConfigs(final @RequestParam ConfigContextEnum context, @RequestParam(required = false) final String descriptorName) {
+    @GetMapping("/{context}")
```

Comment:
Anything that does not uniquely identify a resource by itself should be a RequestParam rather than a PathVariable.